numpy/ma/core.py

```diff
@@ -7096,6 +7096,30 @@ def diag(v, k=0):
     --------
     numpy.diag : Equivalent function for ndarrays.
 
+    Examples
+    --------
+    >>> import numpy as np
+    >>> x = np.array([[11.2, -3.973, 18], [0.801, -1.41, 12], [7, 33, -12]])
+    >>> mask = [row < 0 for row in x]
+    >>> masked_x = np.ma.masked_array(x, mask)
```

Comment:
Just a note, but this should be simplified (not sure if it is copied from elsewhere).  There is no reason for that list comprehension, and e.g. `masked_where` is maybe more readable also.